frontend/src/metabase-lib/lib/metadata/Base.unit.spec.js

```diff
@@ -0,0 +1,53 @@
+import Base from "./Base";
+
+describe("Base", () => {
+  describe("instantiation", () => {
+    it("should set properties from `object` on the Base instance", () => {
+      const instance = new Base({ abc: 123 });
+      expect(instance.abc).toEqual(123);
+    });
+
+    it("should set ALL enumerable properties of `object`, including properties down the prototype chain", () => {
+      const object = {
+        abc: 123,
+      };
+
+      Object.defineProperty(object, "anEnumerableProperty", {
+        enumerable: false,
+        value: false,
+      });
+
+      object.__proto__ = {
+        secretPrototypeValue: true,
+      };
+
+      const instance = new Base(object);
+
+      expect(instance.abc).toEqual(123);
+      expect(instance.secretPrototypeValue).toBe(true);
```

Comment:
Any reason for `toBe(true)` instead of `.toBeTruthy()`?

Reply:
I'm using `.toBe(true)` because it is stricter than `.toBeTruthy()` since it involves no coercion. And there was no thought behind the use of `toEqual` vs `toBe` since they are equivalent for primitive types